src/tests/Microsoft.PowerFx.Interpreter.Tests/RecalcEngineTests.cs

```diff
@@ -588,29 +588,34 @@ private void CheckThroughInterface(IPowerFxEngine engine)
         [Fact]
         public void RecalcEngineLocksConfig()
         {
-            var config = new PowerFxConfig(null);
-            config.SetCoreFunctions(new TexlFunction[0]); // clear builtins
-            config.AddFunction(BuiltinFunctionsCore.Blank);
+            var config = new PowerFxConfig(null)
+            {
+                DefaultFunctions = new SymbolTable() // clear builtins
+            };
+            config.SymbolTable.AddFunction(BuiltinFunctionsCore.Blank);
 
             var recalcEngine = new Engine(config);
 
             var func = BuiltinFunctionsCore.AsType; // Function not already in engine
             Assert.DoesNotContain(func, config.Functions); // didn't get auto-added by engine.
 
+            // We can mutate config after engine is created.
             var optionSet = new OptionSet("foo", DisplayNameUtility.MakeUnique(new Dictionary<string, string>() { { "one key", "one value" } }));
-            Assert.Throws<InvalidOperationException>(() => config.AddFunction(func));
-            Assert.Throws<InvalidOperationException>(() => config.AddOptionSet(optionSet));
+            config.SymbolTable.AddFunction(func);
+            config.SymbolTable.AddEntity(optionSet);

```

Comment:
After this line:

    Assert.Contains(func, config.Functions); // function was added to the config.